controllers/object_controls.go

```diff
@@ -3290,6 +3333,56 @@ func createEmptyDirVolume(volumeName string) corev1.Volume {
 	}
 }
 
+func applyLicensingConfig(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, driverIndex int) {
+	if config.Driver.LicensingConfig == nil || (config.Driver.LicensingConfig.SecretName == "" && config.Driver.LicensingConfig.ConfigMapName == "") {
```

Comment:
Let's remove this if condition from here and move it outside this method. We should only call this method if that `if conditional` evaluates to true